### PR TITLE
Splitting FileIO part 4

### PR DESCRIPTION
**Pull request by norihiro-w**
follow up of #1216 

This PR targets the following files in FileIO, which IMO are more related to applications and shouldn't go to basic modules such as GeoLib and MeshLib. 
- GMSInterface
- PetrelInterface
- SHPInterface
- XmlGspInterface
- XmlLutInterface
- XmlNumInterface
- ImportFileTypes.h

ImportFileTypes.h is moved to Base dir in DE because only DE is using it. Other files are moved to the new directory Applications/FileIO (ApplicationsFileIOLib). Having these in ApplicationsLib/IO is also possible but I'm not sure yet what is a purpose of ApplicationsLib.

### CODE REVIEW

**Comment by bilke** [comment]
Please remove the trailing whitespace in the xsd files to get the green check mark.

**Comment by norihiro-w** [comment]
done

**Comment by endJunction** [comment]
:fast_forward: 

A side note: ApplicationsLib is a questionable one. It was thought to have common code for the applications and utils (like LogogSetup). Suggestion:
LogogSetup.h → BaseLib
ProjectData.* → CLI, because its only used by ogs.cpp
Uncoupled..TimeLoop.h → NumLib or where the time discretization lives now.
LinearSolverSetup.h → MathLib.

File: Applications/DataExplorer/VtkVis/VtkAlgorithmProperties.cpp [hunk 1]
```diff
@@ -22,7 +22,7 @@
 #include <vtkTexture.h>
 
 #include "VtkColorLookupTable.h"
-#include "FileIO/XmlIO/Qt/XmlLutReader.h"
+#include "XmlIO/Qt/XmlLutReader.h"
```
Comment (chleh) [inline Applications/DataExplorer/VtkVis/VtkAlgorithmProperties.cpp]: Is that a full path from the root of the source directory? If not please correct.

File: Applications/DataExplorer/VtkVis/VtkCompositeColormapToImageFilter.cpp [hunk 2]
```diff
@@ -25,7 +25,7 @@
 #include <QFileDialog>
 
 #include "VtkColorLookupTable.h"
-#include "FileIO/XmlIO/Qt/XmlLutReader.h"
+#include "XmlIO/Qt/XmlLutReader.h"
```
Comment (chleh) [inline Applications/DataExplorer/VtkVis/VtkCompositeColormapToImageFilter.cpp]: Likewise.

File: Applications/DataExplorer/mainwindow.cpp [hunk 3]
```diff
@@ -60,14 +60,14 @@
...
+#include "GMSInterface.h"
 #include "MeshLib/IO/FEFLOW/FEFLOWMeshInterface.h"
 #include "MeshLib/IO/Legacy/MeshIO.h"
 #include "MeshLib/IO/readMeshFromFile.h"
 #include "FileIO/GMSHInterface.h"
 #include "FileIO/TetGenInterface.h"
-#include "FileIO/PetrelInterface.h"
-#include "FileIO/XmlIO/Qt/XmlGspInterface.h"
+#include "PetrelInterface.h"
+#include "XmlIO/Qt/XmlGspInterface.h"
```
Comment (chleh) [inline Applications/DataExplorer/mainwindow.cpp]: Likewise.

**Comment by chleh** [comment]
:+1:

**Comment by chleh** [comment]
@endJunction the UncoupledProcessesTimeLoop needs access to the Processes and ProjectData, if I'm right. So it cannot be move to NumLib.

File: Applications/DataExplorer/VtkVis/VtkAlgorithmProperties.cpp (hunk 1 above)
Comment (norihiro-w) [inline Applications/DataExplorer/VtkVis/VtkAlgorithmProperties.cpp]: I won't correct it. DataExplorer doesn't use a full path at many places and those I corrected are exceptions using a full path.

Comment (chleh) [inline Applications/DataExplorer/VtkVis/VtkAlgorithmProperties.cpp]: I see.

**Comment by norihiro-w** [comment]
@endJunction thank you for the explanation. I'll have a detailed look at ApplicationsLib later

**Pull request merged**